### PR TITLE
Fixed projector.

### PR DESCRIPTION
**Pull request by normanjaeckel**

### CODE REVIEW

File: openslides/agenda/projector.py [hunk 1]
```diff
@@ -23,15 +23,10 @@ class ItemListSlide(ProjectorElement):
...
-        if pk is None:
-            # Root list slide.
-            context = {'tree':  self.config_entry.get('tree', False)}
-        else:
+        if pk is not None:
             # Children slide.
             if not Item.objects.filter(pk=pk).exists():
                 raise ProjectorException(_('Item does not exist.'))
-            context = {'id': pk, 'tree':  self.config_entry.get('tree', False)}
-        return context
```
Comment (ostcar) [inline openslides/agenda/projector.py]: The method is namend get_FOO so I would expect a return value. Can you check that this is valid?

File: openslides/agenda/projector.py [hunk 2]
```diff
@@ -65,7 +60,6 @@ def get_context(self):
             raise ProjectorException(_('Id must not be None.'))
         if not Item.objects.filter(pk=pk).exists():
             raise ProjectorException(_('Item does not exist.'))
-        return {'id': pk, 'list_of_speakers': self.config_entry.get('list_of_speakers', False)}
```
Comment (ostcar) [inline openslides/agenda/projector.py]: Here the same

File: openslides/assignments/projector.py [hunk 3]
```diff
@@ -18,15 +18,10 @@ class AssignmentSlide(ProjectorElement):
...
-        if pk is None:
-            # List slide.
-            context = None
-        else:
+        if pk is not None:
             # Detail slide.
             if not Assignment.objects.filter(pk=pk).exists():
                 raise ProjectorException(_('Assignment does not exist.'))
-            context = {'id': pk}
-        return context
```
Comment (ostcar) [inline openslides/assignments/projector.py]: here the same

File: openslides/core/models.py [hunk 4]
```diff
@@ -68,7 +68,7 @@ def elements(self):
         # Parse result
         result = {}
         for key, value in self.config.items():
-            result[key] = value
+            result[key] = value.copy()
```
Comment (ostcar) [inline openslides/core/models.py]: Can you add a comment why the copy is needed?

File: openslides/core/projector.py [hunk 5]
```diff
@@ -16,10 +16,8 @@ class CustomSlideSlide(ProjectorElement):
     name = 'core/customslide'
 
     def get_context(self):
-        pk = self.config_entry.get('id')
-        if not CustomSlide.objects.filter(pk=pk).exists():
+        if not CustomSlide.objects.filter(pk=self.config_entry.get('id')).exists():
             raise ProjectorException(_('Custom slide does not exist.'))
-        return {'id': pk}
```
Comment (ostcar) [inline openslides/core/projector.py]: Here the same

File: openslides/core/serializers.py [hunk 6]
```diff
@@ -9,30 +9,28 @@ class JSONSerializerField(Field):
...
 class ProjectorSerializer(ModelSerializer):
     """
     Serializer for core.models.Projector objects.
     """
+    config = JSONSerializerField(write_only=True)
 
     class Meta:
         model = Projector
-        fields = ('id', 'elements', )
+        fields = ('id', 'config', 'elements', )
```
Comment (ostcar) [inline openslides/core/serializers.py]: Why is config needed again?

File: openslides/motions/projector.py [hunk 7]
```diff
@@ -18,15 +18,10 @@ class MotionSlide(ProjectorElement):
...
-        if pk is None:
-            # List slide.
-            context = None
-        else:
+        if pk is not None:
             # Detail slide.
             if not Motion.objects.filter(pk=pk).exists():
                 raise ProjectorException(_('Motion does not exist.'))
-            context = {'id': pk}
-        return context
```
Comment (ostcar) [inline openslides/motions/projector.py]: Here the same

File: openslides/users/projector.py [hunk 8]
```diff
@@ -14,10 +14,8 @@ class UserSlide(ProjectorElement):
     name = 'users/user'
 
     def get_context(self):
-        pk = self.config_entry.get('id')
-        if not User.objects.filter(pk=pk).exists():
+        if not User.objects.filter(pk=self.config_entry.get('id')).exists():
             raise ProjectorException(_('User does not exist.'))
-        return {'id': pk}
```
Comment (ostcar) [inline openslides/users/projector.py]: Here the same

File: openslides/agenda/projector.py (hunk 1 above)
Comment (normanjaeckel) [inline openslides/agenda/projector.py]: See #1631.

File: openslides/agenda/projector.py (hunk 2 above)
Comment (normanjaeckel) [inline openslides/agenda/projector.py]: See #1631.

File: openslides/assignments/projector.py (hunk 3 above)
Comment (normanjaeckel) [inline openslides/assignments/projector.py]: See #1631.

File: openslides/core/models.py (hunk 4 above)
Comment (normanjaeckel) [inline openslides/core/models.py]: Done.

File: openslides/core/projector.py (hunk 5 above)
Comment (normanjaeckel) [inline openslides/core/projector.py]: See #1631.

File: openslides/core/serializers.py (hunk 6 above)
Comment (normanjaeckel) [inline openslides/core/serializers.py]: It is a write only field now so I can use the serializer in the projector control views.

File: openslides/motions/projector.py (hunk 7 above)
Comment (normanjaeckel) [inline openslides/motions/projector.py]: See #1631.

File: openslides/users/projector.py (hunk 8 above)
Comment (normanjaeckel) [inline openslides/users/projector.py]: See #1631.

**Comment by normanjaeckel** [comment]
All done. Can I rebase and merge?

**Comment by emanuelschuetze** [comment]
@normanjaeckel I tested your branch. There seems to be something wrong if you activate the agenda slide and some item detail slides. The 'prune_elements' post seems to be correctly but the slide will be overridden without clear the old slide. You see it if you test it. (I tested with a fresh db).

**Comment by normanjaeckel** [comment]
Did you also test to reload the page? May be it is an error on the client side (autoupdate).

**Comment by emanuelschuetze** [comment]
You are right! It is a autoupdate problem. After reloading the projector the slide was shown correctly. So client side problem. I'll create a new issue.

**Comment by emanuelschuetze** [comment]
see #1632

**Comment by emanuelschuetze** [comment]
Now, I waiting for merge this pull request that I can rebase my #1619 pull request

**Comment by ostcar** [comment]
@normanjaeckel please rebase and merge

**Pull request merged**